x-pack/plugins/ml/public/application/routing/routes/explorer/state_manager.tsx

```diff
@@ -154,32 +127,33 @@ export const ExplorerUrlStateManager: FC<ExplorerUrlStateManagerProps> = ({
     anomalyExplorerContext.anomalyExplorerCommonStateService.getInfluencerFilterQuery$()
   );
 
-  const loadExplorerDataConfig =
-    explorerState !== undefined
-      ? {
-          lastRefresh,
-          influencersFilterQuery,
-          noInfluencersConfigured: explorerState.noInfluencersConfigured,
-          selectedCells,
-          selectedJobs: explorerState.selectedJobs,
-          tableInterval: tableInterval.val,
-          tableSeverity: tableSeverity.val,
-          viewBySwimlaneFieldName: viewByFieldName,
-        }
-      : undefined;
-
-  useEffect(
-    function updateAnomalyExplorerCommonState() {
-      anomalyExplorerContext.anomalyExplorerCommonStateService.setSelectedJobs(
-        loadExplorerDataConfig?.selectedJobs!
-      );
-    },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [loadExplorerDataConfig]
+  const loadExplorerDataConfig = useMemo(
+    () => ({
+      lastRefresh,
+      influencersFilterQuery,
+      noInfluencersConfigured,
+      selectedCells,
+      selectedJobs,
+      tableInterval: tableInterval.val,
+      tableSeverity: tableSeverity.val,
+      viewBySwimlaneFieldName: viewByFieldName,
+    }),
+    [
+      lastRefresh,
+      influencersFilterQuery,
+      noInfluencersConfigured,
+      selectedCells,
+      selectedJobs,
+      tableInterval,
+      tableSeverity,
+      viewByFieldName,
+    ]
   );
 
   useEffect(() => {
     if (!loadExplorerDataConfig || loadExplorerDataConfig?.selectedCells === undefined) return;
+    // TODO: Find other way to set loading state as it causes unnecessary re-renders - handle it in anomaly_explorer_common_state
```

Comment:
It should be fixed when we decouple `ExplorerState` into individual services for handling, e.g., `chartsData`, `influencers`, etc.